modules/openapi-generator/src/main/resources/go/model.mustache

```diff
@@ -35,9 +35,95 @@ type {{classname}} struct {
 {{#description}}
 	// {{{description}}}
 {{/description}}
-	{{name}} {{#isNullable}}*{{/isNullable}}{{{dataType}}} `json:"{{baseName}}{{^required}},omitempty{{/required}}"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
+	{{name}} *{{{dataType}}} `json:"{{baseName}},omitempty"{{#withXml}} xml:"{{baseName}}{{#isXmlAttribute}},attr{{/isXmlAttribute}}"{{/withXml}}{{#vendorExtensions.x-go-custom-tag}} {{{.}}}{{/vendorExtensions.x-go-custom-tag}}`
+{{#isNullable}}    isExplicitNull{{name}} bool `json:"-"{{#withXml}} xml:"-"{{/withXml}}`{{/isNullable}}
 {{/vars}}
 }
 {{/isEnum}}
+
+{{^isEnum}}
+{{#vars}}
+// Get{{name}} returns the {{name}} field if non-nil, zero value otherwise.
+func (o *{{classname}}) Get{{name}}() {{dataType}} {
+	if o == nil || o.{{name}} == nil {
+		var ret {{dataType}}
+		return ret
+	}
+	return *o.{{name}}
+}
+
+// Get{{name}}Ok returns a tuple with the {{name}} field if it's non-nil, zero value otherwise
+// and a boolean to check if the value has been set.
+func (o *{{classname}}) Get{{name}}Ok() ({{dataType}}, bool) {
+	if o == nil || o.{{name}} == nil {
+		var ret {{dataType}}
+		return ret, false
+	}
+	return *o.{{name}}, true
+}
+
+// Has{{name}} returns a boolean if a field has been set.
+func (o *{{classname}}) Has{{name}}() bool {
+	if o != nil && o.{{name}} != nil {
+		return true
+	}
+
+	return false
+}
+
+// Set{{name}} gets a reference to the given {{dataType}} and assigns it to the {{name}} field.
+func (o *{{classname}}) Set{{name}}(v {{dataType}}) {
+	o.{{name}} = &v
+}
+
+{{#isNullable}}
+// Set{{name}}ExplicitNull (un)sets {{name}} to be considered as explicit "null" value
+// when serializing to JSON (pass true as argument to set this, false to unset)
+// The {{name}} value is set to nil even if false is passed
+func (o *{{classname}}) Set{{name}}ExplicitNull(b bool) {
+	o.{{name}} = nil
+	o.isExplicitNull{{name}} = b
+}
+{{/isNullable}}
+{{/vars}}
+
+func (o {{classname}}) MarshalJSON() ([]byte, error) {
```

Comment:
1. this should belong to another pull, discriminating the change..
2. the custom marshal has to be a opt-in extension..

Reply:
1. As noted in one of the comments above, the custom marshalling is a core of this change without which the rest wouldn't work reasonably.
2. Why?